src/modbus-rtu.c

```diff
@@ -272,32 +272,73 @@ static ssize_t _modbus_rtu_send(modbus_t *ctx, const uint8_t *req, int req_lengt
 #if defined(_WIN32)
     modbus_rtu_t *ctx_rtu = ctx->backend_data;
     DWORD n_bytes = 0;
-    return (WriteFile(ctx_rtu->w_ser.fd, req, req_length, &n_bytes, NULL)) ? (ssize_t)n_bytes : -1;
+    ssize_t rv;
+
+    rv = WriteFile(ctx_rtu->w_ser.fd, req, req_length, &n_bytes, NULL) ? (ssize_t)n_bytes : -1;
+
+    /* if local echo workaround is enabled, try to discard the count of bytes
+     *  we sent ourself */
+    if (ctx_rtu->suppress_echo && n_bytes > 0) {
+        /* we don't use a timeout here since we should have our own caused stuff returned
+         * immediately - and we know the internals of win32_ser_select from code above */
+        if (win32_ser_select(&ctx_rtu->w_ser, (int)n_bytes, NULL) <= 0) {
```

Comment:
will a single call here always be sufficient to get all of our local echo?  we could have written 255 bytes, will they all come back at once?

Reply:
No Windows expect at this side... I'm not sure and had no time yet to setup a libmodbus Windows build/test environment. If somebody have such things at hand, I would be great to test this...

Reply:
Yes it would be great to have Windows dev or testers to help here but it seems a loop is required too (as for Unix).